### PR TITLE
Enable golangci-lint gosec linting, fix issues

### PR DESCRIPTION
**Pull request by dfarrell07**
Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

### CODE REVIEW

File: pkg/spoke/submarineragent/config_controller.go [hunk 1]
```diff
@@ -51,6 +51,7 @@ const (
 	gcpZoneLabel           = "failure-domain.beta.kubernetes.io/zone"
 )
 
+// TODO: Are these hard-coded credntials okay or an issue?
```
Comment (dfarrell07) [inline pkg/spoke/submarineragent/config_controller.go:54]: Taking a second look at this failure, I'm less confident in assuming it's okay. What do you think @skitt?

Comment (skitt) [reply pkg/spoke/submarineragent/config_controller.go:54]: They’re not hard-coded credentials, they’re a reference to a K8s secret which contains credentials.

**Comment by dfarrell07** [comment]
There are two other (non-)issues, but as I understand it this code is going away.

```
pkg/cloud/aws/client/client.go:21:2: G101: Potential hardcoded credentials (gosec)
	accessKeySecretKey   = "aws_secret_access_key"
	^
pkg/cloud/aws/aws.go:34:2: G101: Potential hardcoded credentials (gosec)
	accessKeySecretKey        = "aws_secret_access_key"
	^
```

**Comment by dfarrell07** [comment]
Depends on https://github.com/open-cluster-management/submariner-addon/pull/150

**Review by tpantelis — APPROVED** [review]

**Pull request merged**